src/interpreter.c

```diff
@@ -953,6 +958,11 @@ V7_PRIVATE val_t v7_exec(struct v7 *v7, const char* src) {
 #if 0
   fprintf(stderr, "Eval res: %s .\n", debug);
 #endif
+
+  /* Restore exception environments */
+  memcpy(&v7->jmp_buf, &saved_jmp_buf, sizeof(saved_jmp_buf));
```

Comment:
this function returns in other places, for example an exception is thrown:

```
function testcase() {
    eval("throw 1");
    return prop;
}
testcase();
```

`zsh: abort      ./v7 /tmp/test3.js`

I think a `goto cleanup` is the cleanest way (I'm using this in another branch for restoring `v7->this_object` for `v7_exec_with`)